src/assets.zig

```diff
@@ -10,12 +11,58 @@ const NeverFailingAllocator = main.utils.NeverFailingAllocator;
 var arena: main.utils.NeverFailingArenaAllocator = undefined;
 var arenaAllocator: NeverFailingAllocator = undefined;
 var commonBlocks: std.StringHashMap(ZonElement) = undefined;
+var commonBlocksMigrations: std.StringHashMap(ZonElement) = undefined;
 var commonBiomes: std.StringHashMap(ZonElement) = undefined;
 var commonItems: std.StringHashMap(ZonElement) = undefined;
 var commonTools: std.StringHashMap(ZonElement) = undefined;
 var commonRecipes: std.StringHashMap(ZonElement) = undefined;
 var commonModels: std.StringHashMap([]const u8) = undefined;
 
+
+pub fn init() void {
+	biomes_zig.init();
+	blocks_zig.init();
+
+	arena = .init(main.globalAllocator);
+	arenaAllocator = arena.allocator();
+	commonBlocks = .init(arenaAllocator.allocator);
+	commonBlocksMigrations = .init(arenaAllocator.allocator);
+	commonItems = .init(arenaAllocator.allocator);
+	commonTools = .init(arenaAllocator.allocator);
+	commonBiomes = .init(arenaAllocator.allocator);
+	commonRecipes = .init(arenaAllocator.allocator);
+	commonModels = .init(arenaAllocator.allocator);
+
+	readAssets(
+		arenaAllocator,
+		"assets/",
+		&commonBlocks,
+		&commonBlocksMigrations,
+		&commonItems,
+		&commonTools,
+		&commonBiomes,
+		&commonRecipes,
+		&commonModels,
+	);
+
+	// Migrations have to be registered before the world is loaded so they can be
+	// applied to the world during load.
+	migrations_zig.registerBlockMigrations(&commonBlocksMigrations);
```

Comment:
This is unnecessary, since you already register them below in the `loadWorldAssets`, this is registering them twice.